elements/lisk-cryptography/src/bls_lib/lib.ts

```diff
@@ -45,9 +45,31 @@ export const blsKeyValidate = (pk: Buffer): boolean => {
 // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.3
 export const blsKeyGen = (ikm: Buffer): Buffer => Buffer.from(SecretKey.fromKeygen(ikm).toBytes());
 
+// check that the secret key generated is non-zero modulo the order of the elliptic curve.
+export const isSecretKeyNonZeroModEC = (secretKey: SecretKey): boolean => {
+	// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#section-4.2.1
+	const curveOrder = '0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001';
```

Comment:
I couldn't tell why this value is used for checking modulo from the spec

Reply:
Oh maybe I should point the link to the exact value. the [order](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#section-4.2.1-11.4) is `r = 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001`. Thanks to @AndreasKendziorra who pointed this out to me.

Reply:
can be resolved from my side